pages/DSNP/Graph.md

```diff
@@ -2,29 +2,124 @@
 
 This specification describes the social network graph and how it is represented in the protocol.
 
-In this context a [social graph](https://en.wikipedia.org/wiki/Social_graph) means a graph that represents social relations between entities.
+In this context a [social graph](https://en.wikipedia.org/wiki/Social_graph) means a graph that represents social relationships between entities.
 
-The DSNP graph represents nodes as [DSNP User Ids](Identifiers.md#dsnp-user-id).
+## Nodes
 
-## Public Graph
+Users are represented in the DSNP graph by nodes uniquely identified by their [DSNP User Ids](Identifiers.md#dsnp-user-id).
+Relationships between users are the graph edges.
 
-The public graph is anchored in state transitions recorded in [Graph Change Announcements](Types/GraphChange.md).
-Playing all state transitions in [order](Announcements.md#ordering-announcements) will generate the current state of the public graph.
+## Edges
 
-## Non-Normative
+Graph edges that originate from a user are stored as DSNP [User Data](UserData.md) associated with a given user.
 
-### Private Graph
+Each graph edge consists of the following data:
+* DSNP User Id (64-bit unsigned integer)
```

Comment:
The signed/unsigned is strange here. It feels odd to have a uint64 everywhere but here.

Reply:
This is a compromise driven by the lack of a native unsigned type in Avro. The alternative would be to store as a fixed 8-byte value. This is better aligned with actual User Ids, but doesn't allow the Avro serialization to do any integer packing, which pushes responsibility to the compression codec, which has less context to work with.

Comparison of 300 GraphEdge entries using evenly distributed User Ids from 0 to 300,000,000:
| User Id data type | Avro binary serialization size (bytes) | Size after applying Brotli compression (bytes) |
| --- | --- | --- |
| `long` | 2,247 | 2,167 |
| `fixed` | 3,900 | 2,841 |

`fixed` is consistently about 30% larger after compression.